src/hooks/sounds.tsx

```diff
@@ -35,35 +37,56 @@ export interface ExposedDataWithPlay extends ExposedData {
 
 export interface CustomSoundsState {
   soundConfigs: SoundConfigMap;
+  isLoading: boolean;
 }
 
 export const initialValue: CustomSoundsState = {
   soundConfigs: {},
+  isLoading: false,
 };
 
 export const CustomSoundsContext = createContext<CustomSoundsState>(
   initialValue
 );
 
-export const CustomSoundsProvider: React.FC = ({ children }) => {
+export type CustomSoundsType = {
+  loadingComponent: React.ReactNode;
+  waitTillConfigLoaded: boolean;
+};
+
+export const CustomSoundsProvider: React.FC<CustomSoundsType> = ({
+  loadingComponent,
+  waitTillConfigLoaded,
+  children,
+}) => {
   const [soundConfigs, setSoundConfigs] = useState<SoundConfigMap>(
     initialValue.soundConfigs
   );
+  const [isLoading, setIsLoading] = useState<boolean>(initialValue.isLoading);
+  const { user } = useUser();
+  const userId = user?.uid;
 
   // Fetch the sound configs data on first load
   useEffect(() => {
-    fetchSoundConfigs().then((soundConfigs) => {
-      setSoundConfigs(soundConfigs);
-    });
-  }, []);
+    if (!userId) return;
+    setIsLoading(true);
+    fetchSoundConfigs()
+      .then((soundConfigs) => setSoundConfigs(soundConfigs))
+      .finally(() => setIsLoading(false));
+  }, [userId]);
 
   const providerData = useMemo(
     () => ({
       soundConfigs,
+      isLoading,
     }),
-    [soundConfigs]
+    [soundConfigs, isLoading]
   );
 
+  if (isLoading && waitTillConfigLoaded) {
+    return <>{loadingComponent}</>;
+  }
```

Comment:
I think this might be better 'inside' the `CustomSoundsContext.Provider` below. Possibly just as a ternary?

Something like:

```typescript
return (
    <CustomSoundsContext.Provider value={providerData}>
      {waitTillConfigLoaded && isLoading ? loadingComponent : children}
    </CustomSoundsContext.Provider>
  );
```

Reply:
fixed